src/pip/_internal/cli/cmdoptions.py

```diff
@@ -539,6 +544,15 @@ def only_binary() -> Option:
     dest="platforms",
     metavar="platform",
     action="append",
+    choices=[
+        "any",
+        "manylinux1_x86_64",
+        "manylinux1_i386",
+        "manylinux2014_aarch64",
+        "win_amd64",
+        "macosx_10_9_x86_64",
+        "macosx_11_0_arm64",
+    ],
```

Comment:
Why only these?

Reply:
Is there any resource to list all platforms? I just search some one, but it is not all.

And some checks are fail because of similar reason. Such as `--implementation fakepy`. fakepy seems not to be one of implementation.